app/controllers/products_controller.rb

```diff
@@ -0,0 +1,91 @@
+class ProductsController < ApplicationController
+	before_action :require_login, only: [:new, :create, :show_seller_products, :edit, :update, :retire, :destroy]
+
+	def index
+		@products = Product.where(retired: false)
+	end
+
+	def show
+		@product = Product.find(params[:id])
+		@quantity = @product.quantity
+		@quantity_numbers = (1..@quantity).to_a
+	end
+
+	def show_seller_products
+		@user = User.find(current_user)
+		@products = @user.products
+	end
+
+	def show_merchant
+		@user = User.find(params[:id])
+		@products = @user.products.where(retired: false)
+	end
+
+	def show_animal
+		@products = Product.where(animal: params[:animal]).where(retired: false)
+	end
+
+	def show_category
+		@products = Product.where(category: params[:category]).where(retired: false)
+	end
+
+	def new
+		@user = User.find(params[:id])
+		@product = @user.products.new
+		@category = Product.uniq.pluck(:category)
+		@animal = Product.uniq.pluck(:animal)
+		@category << "Create a Category"
+	end
```

Comment:
This is a great way to make sure your dropdowns always stay up to date! Love it.